opm/material/fluidmatrixinteractions/EclHysteresisTwoPhaseLawParams.hpp

```diff
@@ -644,8 +644,8 @@ class EclHysteresisTwoPhaseLawParams : public EnsureFinalized
             Scalar krnMdcDrainage = EffLawT::twoPhaseSatKrn(drainageParams(), krnSwMdc_);
             Scalar SwKrnMdcImbibition = EffLawT::twoPhaseSatKrnInv(imbibitionParams(), krnMdcDrainage);
             deltaSwImbKrn_ = SwKrnMdcImbibition - krnSwMdc_;
-            assert(std::abs(EffLawT::twoPhaseSatKrn(imbibitionParams(), krnSwMdc_ + deltaSwImbKrn_)
-                            - EffLawT::twoPhaseSatKrn(drainageParams(), krnSwMdc_)) < 1e-8);
+            //assert(std::abs(EffLawT::twoPhaseSatKrn(imbibitionParams(), krnSwMdc_ + deltaSwImbKrn_)
+            //                - EffLawT::twoPhaseSatKrn(drainageParams(), krnSwMdc_)) < 1e-8);
         }
```

Comment:
Do we want assert here? I need to change it to 1e-7 to make the new test pass.

Reply:
> Do we want assert here?

Probably not.  `Assert()` is never a good check for anything other than logic errors and the check compiles away to nothing in release builds which define `NDEBUG`.  On the other hand, the condition itself&ndash;that the drainage curve does not deviate too much from the imbibition curve at the switch point&ndash;might be reasonable.  If it is, then this should be a proper check with some way of signalling the calling code that not all is well&ndash;e.g., an exception.

Reply:
I removed the assert, and instead add the check to the test. This assert test that the twoPhaseSatKrnInv is doing its job. I dont see the point of checking that every time the hysteresis code is evaluated.